packages/transformers/babel/src/babel7.js

```diff
@@ -50,6 +57,45 @@ export default async function babel7(
     );
   } else {
     res = await babel.transformAsync(await asset.getCode(), config);
+    if (res.ast) {
+      let map = await asset.getMap();
+      if (map) {
+        // remap ast to original mappings
+        // This improves sourcemap accuracy and fixes sourcemaps when scope-hoisting
+        traverse(res.ast.program, {
+          enter(path) {
+            if (path.node.loc) {
+              if (path.node.loc?.start) {
+                // TODO: Check if babel ast uses 1 or zero based indexes and match to our indexing system...
+                let mapping = map.findClosestMapping(
+                  path.node.loc.start.line,
+                  path.node.loc.start.column,
+                );
+
+                if (mapping?.original) {
+                  // $FlowFixMe
+                  path.node.loc.start.line = mapping.original.line;
+                  // $FlowFixMe
+                  path.node.loc.start.column = mapping.original.column;
+
+                  if (path.node.loc?.end) {
```

Comment:
Not entirely sure yet how to handle the end locations...

These should probably be the mapping after the closest mapping if the sourcefile is the same, however to make this efficient we should maybe add this logic to the sourcemap library?